pkg/vetter/util/mtlspolicy/authPolicy_test.go

```diff
@@ -160,56 +193,475 @@ var (
 var _ = Describe("LoadAuthPolicies", func() {
 	It("should load policies", func() {
 		loaded, err := LoadAuthPolicies([]*authv1alpha1.Policy{
-			apDefaultOn,
-			apFooOn,
-			apFooOff,
-			apFooBarOn,
-			apFooPortsBarOn,
+			nsbarNs_On,
+			nsDefault_apFoo_On,
+			nsDefault_apFoo_Off,
+			nsDefault_apFoo_apBar_On,
+			nsDefault_apFooPorts_apBar_On,
 		})
-		Expect(err).To(Succeed())
+		Expect(err).To(BeNil())
+		loaded.LoadMeshPolicy([]*authv1alpha1.MeshPolicy{meshDefaultOn})
+
 		foo := Service{Namespace: "default", Name: "foo"}
 		bar := Service{Namespace: "default", Name: "bar"}
-		Expect(loaded.ByNamespace("barNs")).To(Equal([]*authv1alpha1.Policy{apDefaultOn}))
+
+		Expect(loaded.ByMesh()).To(Equal([]*authv1alpha1.MeshPolicy{meshDefaultOn}))
+		Expect(loaded.ByNamespace("barNs")).To(Equal([]*authv1alpha1.Policy{nsbarNs_On}))
 		Expect(loaded.ByNamespace("default")).To(Equal([]*authv1alpha1.Policy{}))
 
 		Expect(loaded.ByName(foo)).To(Equal([]*authv1alpha1.Policy{
-			apFooOn,
-			apFooOff,
-			apFooBarOn,
-			// no apFooPortsBarOn because that is only for foo:8443, not foo
+			nsDefault_apFoo_On,
+			nsDefault_apFoo_Off,
+			nsDefault_apFoo_apBar_On,
+			// no nsDefault_apFooPorts_apBar_On because that is only for foo:8443, not foo
 		}))
 		Expect(loaded.ByName(bar)).To(Equal([]*authv1alpha1.Policy{
-			apFooBarOn,
-			// apFooPortsBarOn because that is for bar (not bar:8443)
-			apFooPortsBarOn,
+			nsDefault_apFoo_apBar_On,
+			// nsDefault_apFooPorts_apBar_On because that is for bar (not bar:8443)
+			nsDefault_apFooPorts_apBar_On,
 		}))
 
-		Expect(loaded.ByPort(foo, 8443)).To(Equal([]*authv1alpha1.Policy{apFooPortsBarOn}))
+		Expect(loaded.ByPort(foo, 8443)).To(Equal([]*authv1alpha1.Policy{nsDefault_apFooPorts_apBar_On}))
 		Expect(loaded.ByPort(foo, 1000)).To(Equal([]*authv1alpha1.Policy{}))
 		Expect(loaded.ByPort(bar, 8443)).To(Equal([]*authv1alpha1.Policy{}))
 	})
 })
 
-var _ = Describe("GetClusterPortPolicies", func() {
-	It("should return port policies from the loaded AuthPolicies struct", func() {
-		loaded, err := LoadAuthPolicies([]*authv1alpha1.Policy{
-			apDefaultOn,
-			apFooOn,
-			apFooOff,
-			apFooBarOn,
-			apFooPortsBarOn,
+var _ = Describe("AuthPolicyIsMtls", func() {
+	It("should evaluate no-mtls-peer as false", func() {
+		Expect(AuthPolicyIsMtls(nsDefault_apFoo_Off)).To(Equal(MTLSSetting_DISABLED))
+	})
+	It("should evaluate empty-mtls-peer as true", func() {
+		Expect(AuthPolicyIsMtls(nsDefault_apFooPorts_apBar_On)).To(Equal(MTLSSetting_ENABLED))
+	})
+})
+
+var _ = Describe("test AuthPolicyIsMtls() through its callers", func() {
```

Comment:
I like this suite but I would pick a different name.  In other words, if you refactored and TLSDetailsByNamespace no longer called AuthPolicyIsMtls you would still want the unit test below.

Maybe just "TLSDetails"?